backup_restore_comparison.py

```diff
@@ -0,0 +1,338 @@
+        size_in_bytes /= 1024
+
+
+class ManagerBackupRestoreConcurrentTests(ManagerTestFunctionsMixIn):
+    snapshot_list_lock = threading.Lock()
+    snapshot_ids = {}
+    node_sstables = {}
+    node_backup_size = {}
+    base_prefix = ""
+
+    def report_to_argus(self, report_type: ManagerReportType, data: dict, label: str):
+        if report_type == ManagerReportType.READ:
+            table = ManagerBackupReadResult(sut_timestamp=mgmt.get_scylla_manager_tool(
+                manager_node=self.monitors.nodes[0]).sctool.client_version_timestamp)
+        elif report_type == ManagerReportType.BACKUP:
+            table = ManagerBackupBenchmarkResult(sut_timestamp=mgmt.get_scylla_manager_tool(
+                manager_node=self.monitors.nodes[0]).sctool.client_version_timestamp)
+        else:
+            raise ValueError("Unknown report type")
+
+        for key, value in data.items():
+            table.add_result(column=key, value=value, row=label, status=Status.UNSET)
+        submit_results_to_argus(self.test_config.argus_client(), table)
+
+    def create_backup_and_report(self, mgr_cluster, label: str):
+        InfoEvent(message='Starting `rclone` based backup').publish()
+        task = mgr_cluster.create_backup_task(location_list=self.locations, rate_limit_list=["0"])
+
+        backup_status = task.wait_and_get_final_status(timeout=72000)
+        assert backup_status == TaskStatus.DONE, "Backup upload has failed!"
+
+        backup_report = {
+            "Size": parse_backup_size(mgr_cluster, task.id),
+            "Time": int(task.duration.total_seconds()),
+        }
+        self.report_to_argus(ManagerReportType.BACKUP, backup_report, label)
+        return task
+
+    def create_restore_and_report(self, mgr_cluster, snapshot_tag, label: str):
+        self.set_balancing(False)
+
+        task = self.restore_backup_with_task(mgr_cluster=mgr_cluster, snapshot_tag=snapshot_tag,
+                                             timeout=72000, restore_data=True)
+
+        backup_status = task.wait_and_get_final_status(timeout=72000)
+        assert backup_status == TaskStatus.DONE, "Backup upload has failed!"
+
+        backup_report = {
+            "Size": parse_backup_size(mgr_cluster, task.id),
+            "Time": int(task.duration.total_seconds()),
+        }
+        self.report_to_argus(ManagerReportType.BACKUP, backup_report, label)
+        self.set_balancing(True)
+        return task
+
+    def run_read_stress_and_report(self, label):
+        stress_queue = {"read": [], "write": []}
+
+        for command in self.params.get('stress_cmd'):
+            if " write " in command:
+                stress_queue["write"].append(self.run_stress_thread(command))
+            elif " read " in command:
+                stress_queue["read"].append(self.run_stress_thread(command))
+            else:
+                raise ValueError("Unknown stress command")
+
+        def get_stress_averages(queue):
```

Comment:
I am not sure you can average results from stresses and get a valid 99th percentile. Is this how we do it in the rest of the performance tests? Would itbe possible to reuse some of the functions/decorators from there to keep it consistent?

Reply:
> Would it be possible to reuse some of the functions/decorators from there to keep it consistent?

Can you point me to?

Reply:
I would take a look at https://github.com/scylladb/scylla-cluster-tests/blob/eab891577d32d160976c971f12b06cb9de4c4c81/performance_regression_test.py#L48